CMakeLists.txt

```diff
@@ -144,8 +144,12 @@ GLOBAL_SET(KOKKOS_XCOMPILER_OPTIONS)
 # We need to append text here for making sure TPLs
 # we import are available for an installed Kokkos
 GLOBAL_SET(KOKKOS_TPL_EXPORTS)
-# this could probably be scoped to project
+# KOKKOS_DEPENDENCE is used by kokkos_launch_compiler
 GLOBAL_SET(KOKKOS_COMPILE_DEFINITIONS KOKKOS_DEPENDENCE)
+# MSVC never goes through kokkos_launch_compiler
+IF(NOT MSVC)
+    GLOBAL_APPEND(KOKKOS_LINK_OPTIONS -DKOKKOS_DEPENDENCE)
+ENDIF()
```

Comment:
Can please again elaborate on the issue this is supposed to fix?

Reply:
Well A, that flag wasn't showing up in the exported link flags. And B, the new system should always have that present, not just when CUDA is used.

Reply:
I believe the reason it wasn't showing up in the exported link flags might have been that it was getting added to the `KOKKOS_LINK_OPTIONS` after `KOKKOS_LINK_OPTIONS` was exported, so it was getting applied in the build tree but the install tree export didn't have it. Didn't dive into too much bc re-adding it here is basically what we need to do anyway to make sure it always there.